rust/otap-dataflow/crates/otlp/src/otlp_batch_processor.rs

```diff
@@ -10,134 +10,154 @@ use otap_df_engine::error::Error;
 use otap_df_engine::local::processor::{EffectHandler, Processor};
 use otap_df_engine::message::{ControlMsg, Message};
 use prost::Message as ProstMessage;
+use std::borrow::Cow;
 use std::time::{Duration, Instant};
 
 /// Trait for hierarchical batch splitting
 ///
 /// This trait is used to split a batch into a vector of smaller batches, each with at most `max_batch_size`
 /// leaf items, preserving all resource/scope/leaf (span/metric/logrecord) structure.
 pub trait HierarchicalBatchSplit: Sized {
-    fn split_into_batches(self, max_batch_size: usize) -> Vec<Self>;
+    fn split_into_batches(self, max_batch_size: usize) -> Result<Vec<Self>, Error<OTLPData>>;
 }
 
 /// TODO: Use the pdata/otlp support library, rewrite this function to be generic over PData as that library develops
 impl HierarchicalBatchSplit for ExportTraceServiceRequest {
-    fn split_into_batches(mut self, max_batch_size: usize) -> Vec<Self> {
-        let mut batches = Vec::new();
+    fn split_into_batches(mut self, max_batch_size: usize) -> Result<Vec<Self>, Error<OTLPData>> {
+        if max_batch_size == 0 {
+            return Err(Error::ProcessorError {
+                processor: Cow::Borrowed("HierarchicalBatchSplit::ExportTraceServiceRequest"),
+                error: "max_batch_size must be greater than zero".into(),
+            });
+        }
```

Comment:
nit: I would expect to see a check like this in the start() method, or whenever reconfiguration happens.